### ISSUE TITLE
Panic decrypting FRED2.MZX

### ISSUE THREAD

**Issue by jdm** [jdm/libmzx#51]
https://www.digitalmzx.com/show.php?id=342
```
thread 'main' panicked at 'attempt to subtract with overflow', /Users/jdm/src/libmzx/src/lib.rs:2347:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/81dc88f88f92ba8ad7465f9cba10c12d3a7b70f1/library/std/src/panicking.rs:475
   1: core::panicking::panic_fmt
             at /rustc/81dc88f88f92ba8ad7465f9cba10c12d3a7b70f1/library/core/src/panicking.rs:85
   2: core::panicking::panic
             at /rustc/81dc88f88f92ba8ad7465f9cba10c12d3a7b70f1/library/core/src/panicking.rs:50
   3: libmzx::decrypt_and_fix_offset
             at /Users/jdm/src/libmzx/src/lib.rs:2347
   4: libmzx::load_world
             at /Users/jdm/src/libmzx/src/lib.rs:2468
   5: capture::run
             at ./capture/src/main.rs:63
   6: capture::main
             at ./capture/src/main.rs:220
   7: core::ops::function::FnOnce::call_once
             at /Users/jdm/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/ops/function.rs:233
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by jdm** [comment]
Also https://www.digitalmzx.com/show.php?id=965

**Comment by jdm** [comment]
Fixed by a1a53cdb09261db7e4cba506e631dec7a0d92f1e.
